lib/spack/spack/util/web.py

```diff
@@ -203,14 +203,9 @@ def warn_no_ssl_cert_checking():
              "your Python to enable certificate verification.")
 
 
-def push_to_url(local_path, remote_path, **kwargs):
+def push_to_url(local_file_path, remote_path, **kwargs):
```

Comment:
I think I understand how this change addresses #13404, but it seems to imply that there is something broken with how `file://` URLs are parsed.  It looks like urllib is stripping out the part of the filename that looks like query arguments, and that will still be a problem if we ever have to work with similar file paths, either as part of something internal to Spack (as in this case), or if a user ever hands us such a file path.

I'd like to take a closer look at this and possibly follow up with @scheibelp if I have any questions.